src/main/java/im/status/keycard/SecureChannel.java

```diff
@@ -176,6 +199,66 @@ public void mutuallyAuthenticate(APDU apdu) {
     respond(apdu, len, ISO7816.SW_NO_ERROR);
   }
 
+  /**
+   * Processes the LOAD_CERTS command. Copies the APDU buffer into `certs`.
+   * This function expects a DER signature and may only be called once.
+   * @param apdu the JCRE-owned APDU object.
+   */
+  public void loadCert(APDU apdu) {
+    byte[] apduBuffer = apdu.getBuffer();
+
+    final short caPubOffset = (short) ISO7816.OFFSET_CDATA;
+    final short certOffset = (short) (caPubOffset + PUBKEY_LEN);
+    final short certVerifyLen = (short)(CERTIFICATE_LEN - ECDSA_MAX_LEN);
+
+    if (idCertStatus != ID_CERTIFICATE_EMPTY) {
+      // Card cert may only be set once and never overwritten
+      ISOException.throwIt(ISO7816.SW_COMMAND_NOT_ALLOWED);
+    }
+
+    // Make sure the received certificate is appropriate length
+    if (apduBuffer[ISO7816.OFFSET_LC] != (byte) (PUBKEY_LEN + CERTIFICATE_LEN)) {
+      ISOException.throwIt(ISO7816.SW_DATA_INVALID);
+    }
+
+    JCSystem.beginTransaction();
+    // Save the certificate
+    Util.arrayCopy(apduBuffer, certOffset, idCertificate, (short) 0, CERTIFICATE_LEN);
+
+    // Lock the certificate
+    idCertStatus = ID_CERTIFICATE_LOCKED;
+    JCSystem.commitTransaction();
```

Comment:
Don't you need to `setOutgoingAndSend`?

Reply:
Outdated comment again. The latest code has this fixed

Reply:
I asked you about this, but I thought you told me I do not need to `setOutgoingAndSend` if I have not put any data in the APDU response. Did I misunderstand?

Reply:
Right... that may be correct. I actually don't know. Can you find an APDU that Status uses where they don't return anything? If they are neglecting the `setOutgoingAndSend`, you can do that too.

Reply:
Yeah, several of their APDUs do that. e.g. `removeKey()`